core/server/master/src/main/java/alluxio/master/file/meta/InodeTree.java

```diff
@@ -765,10 +766,13 @@ public CreatePathResult createPath(RpcContext rpcContext, LockedInodePath inodeP
             currentInodeDirectory, name);
 
         // if the parent has default ACL, copy that default ACL as the new directory's default
-        // and access acl.
+        // and access acl, taking into account the creation umask of the dir.
+        short mode = options.isMetadataLoad() ? Mode.createFullAccess().toShort()
```

Comment:
can you add comment on how this depends on loading metadata?

also, is this acting like the umask? If so, can we call it umask?